javatools/src/main/java/org/xvm/tool/XdkResolvingLauncher.java

```diff
@@ -0,0 +1,125 @@
+package org.xvm.tool;
+
+import org.xvm.tool.Launcher.LauncherException;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Launcher that figures out if it's part of an XDK installation. If this
+ * is not the case, it fails. If it is, it should resolve the javatools xtc modules
+ * and the xdk modules from its self-discovered location and prepend those as
+ * arguments to the "real" launcher. This makes it possible to replace the
+ * native binary launch exes with auto generated application plugin scripts.
+ * These, in turn, can be turned into binary launchers, should we want to.
+ * This frees up a lot of rather complex binary handling in our bits, as well
+ * as symlinks that are required for this location check to work in the current
+ * solution.
+ * <p>
+ * The logic implemented in this file is identical to the logic used by our
+ * legacy binary launchers for identifying and resolving if we are executing
+ * from and XTC installation, and the locations of its core libraries required
+ * to run are present.
+ * <p>
+ * The application plugin launch config will look something like this:
+ *   + applicationDefaultJvmArgs = "-ea" (as in the current native launchers)
+ *   + applicationDistribution = the xdk/build/install directory, or rather, the outputs of installDist
```

Comment:
Do we really have to se this? Shouldn't this be part of the convention?

Reply:
It is totally part of the convention, at the moment the CR is rather much "work in progress and note to self".  Rest assured t will be cleaned up.  Btw - since the application plugin only assumes one application config per project, I'm probably just going to add two createStartScript type tasks to xdk/build.gradle.kts and have the distribution generation depend on those. The distribution can copy them to the bin dir . Right now the distribution only copies the three platform specific exe launchers to libexec/bin, and doesn't put anything in the upper bin directory for the distribution. This is where I want to place the start scripts, which should be trivial to extend the distribution configuration to do.    That, if I understand it correctly, supersedes both the need for symlinking anything anywhere, as well as requiring an installed home-brew distribution with the launcher links in the right place, where we overwrite arbitrary parts of that tree to "refresh" a local install .

Reply:
Working as fast as I can, but I got interrupted by debugability questions for a bit so I am a bit behind schedule, but not much left now hopefully.